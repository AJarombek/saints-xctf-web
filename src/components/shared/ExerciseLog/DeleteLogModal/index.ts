import DeleteLogModal from './DeleteLogModal';

export default DeleteLogModal;
