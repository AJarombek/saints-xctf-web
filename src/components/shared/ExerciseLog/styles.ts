/**
 * JSS styles for the ExerciseLog component.
 * @author Andrew Jarombek
 * @since 7/26/2020
 */

import Colors, {FeelColors} from "../../../styles/colors";
import {FontMixins} from "../../../styles/mixins";
import color from "color";

export default {
  exerciseLog: {
    backgroundColor: ({feel}: {feel: number}) => FeelColors[feel - 1],
    border: '2px solid #888',
    borderRadius: '3px',
    padding: '7px',
    margin: '10px 0',
    boxShadow: '0 2px 4px 0 rgba(0, 0, 0, 0.2)',
    position: 'relative'
  },
  headerSection: {
    display: 'flex'
  },
  titles: {
    display: 'block'
  },
  titleLink: {
    ...FontMixins.robotoSlab(),
    color: '#333',
    textDecoration: 'none',
    fontSize: 16
  },
  title: {
    ...FontMixins.robotoSlab(),
    textDecoration: 'underline',
    fontSize: '16px',
  },
  metadata: {
    display: 'block',
    margin: '0 0 0 auto'
  },
  date: {
    ...FontMixins.robotoBold(),
    fontSize: '14px',
    margin: 0,
  },
  type: {
    ...FontMixins.robotoSlab(),
    fontSize: '14px',
    fontColor: '#666',
    margin: 0,
    textAlign: 'right',
  },
  bodySection: {
    marginTop: '20px',
  },
  dataFields: {
    '& > p': {
      ...FontMixins.robotoSlab(),
      fontSize: '14px',
      margin: 0,
    }
  },
  description: {
    marginTop: '10px',
    display: 'inline-block',
    ...FontMixins.robotoSlab(),
    fontSize: '14px',

    '& > a': {
      display: 'inline-block',
      ...FontMixins.robotoSlabBold(),
      color: '#444',
      transition: 'color 0.4s ease',

      '&:hover': {
        color: color(Colors.spotPaletteBrown).darken(0.2).hex(),
      }
    },
  },
  commentSection: {
    margin: '10px 0 0 0',
  },
  options: {
    position: 'absolute',
    width: 'calc(100% - 14px)',
    display: 'flex'
  },
  optionsButtons: {
    margin: '0 0 0 auto',
    display: 'flex'
  },
  optionsButton: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',

    borderRadius: '50%',
    border: 'none',
    backgroundColor: Colors.spotPaletteCream,
    boxShadow: '0 2px 4px 0 rgba(0, 0, 0, 0.1)',
    cursor: 'pointer',
    height: 30,
    width: 30,
    margin: '0 2px',

    '& > p': {
      fontSize: 20,
      color: '#555'
    },

    '&:focus': {
      outline: 'none',
    }
  },
  deleteOptionsButton: {
    backgroundColor: color(Colors.sxctfRed).lighten(0.25).hex(),

    '& > p': {
      color: '#e6e6e6'
    }
  },
  optionsIcon: {
    ...FontMixins.elegantIcons(),
  },
  deleteModal: {
    display: 'flex',
    flexDirection: 'column',
    margin: '20px 40px',

    '& > p': {
      ...FontMixins.robotoSlab(),

      '& b': {
        ...FontMixins.robotoSlabBold(),
      }
    }
  },
  deleteModalButtons: {
    display: 'flex',
    marginLeft: 'auto',
    marginTop: 20,

    '& button': {
      display: 'flex',
      alignItems: 'center'
    },

    '& > .aj-contained-button, & > .aj-outlined-button': {

      '& p': {
        margin: 0
      }
    },

    '& > .aj-contained-button': {
      marginRight: 10
    }
  },
  disabledDeleteButton: {
    backgroundColor: '#e6e6e6 !important',

    '& p': {
      color: '#555'
    }
  },
  deleteLogSpinner: {
    marginLeft: 25,
    marginRight: 10,

    '&:before': {
      border: `solid 4px ${color(Colors.spotPaletteBrown).lighten(0.75).hex()}`,
      borderTopColor: Colors.spotPaletteBrown,
    }
  }
};
