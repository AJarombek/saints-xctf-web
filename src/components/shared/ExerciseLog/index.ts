import ExerciseLog from "./ExerciseLog";

export default ExerciseLog;
