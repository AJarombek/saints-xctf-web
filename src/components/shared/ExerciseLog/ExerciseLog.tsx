/**
 * ExerciseLog component which displays a published exercise log, often in a log feed.
 * @author Andrew Jarombek
 * @since 7/26/2020
 */

import React from 'react';
import {createUseStyles} from "react-jss";
import styles from "./styles";
import {Log, NewComments, User} from "../../../redux/types";
import {Link} from "react-router-dom";
import moment from "moment";
import Comments from "../Comments/Comments";
import {shortenTime} from "../../../utils/logs";

interface IProps {
    log: Log;
    postComment: (logId: number, username: string, first: string, last: string, content: string) => void;
    addComment: (logId: number, content: string, username: string, first: string, last: string,
                 filterBy: string, bucket: string, page: number, index: number) => void;
    newComments: NewComments;
    user: User;
    page: number;
    filterBy: string;
    bucket: string;
    index: number;
}

const useStyles = createUseStyles(styles);

const ExerciseLog: React.FunctionComponent<IProps> = ({
    log,
    postComment,
    addComment,
    newComments,
    user,
    page,
    filterBy,
    bucket,
    index
}) => {
    const classes = useStyles({ feel: log?.feel });

    return (
        <div className={classes.exerciseLog}>
            <div className={classes.headerSection}>
                <div className={classes.titles}>
                    <Link to="/user" className={classes.titleLink}>{log.first} {log.last}</Link>
                    <h6 className={classes.title}>{log.name}</h6>
                </div>
                <div className={classes.metadata}>
                    <p className={classes.date}>{moment(log.date).format('MMM. Do, YYYY')}</p>
                    <p className={classes.type}>{log.type.toUpperCase()}</p>
                </div>
            </div>
            <div className={classes.bodySection}>
                <div className={classes.dataFields}>
                    {!!log.location && <p>Location: {log.location}</p>}
                    {!!log.distance && <p>{log.distance} {log.metric}</p>}
                    {!!log.time && log.time !== "0:00:00" && (
                        <p>{shortenTime(log.time)} ({shortenTime(log.pace)}/mi)</p>
                    )}
                </div>
                <div className={classes.description}>
                    <p>{log.description}</p>
                </div>
            </div>
            <div className={classes.commentSection}>
                <Comments
                    comments={log.comments}
                    feel={log.feel}
                    postComment={postComment}
                    addComment={addComment}
                    newComments={newComments}
                    logId={log.log_id}
                    user={user}
                    page={page}
                    filterBy={filterBy}
                    bucket={bucket}
                    index={index}
                />
            </div>
        </div>
    );
};

export default ExerciseLog;
