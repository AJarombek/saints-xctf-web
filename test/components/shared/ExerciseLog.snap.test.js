/**
 * Snapshot test for the ExerciseLog component.
 * @author Andrew Jarombek
 * @since 8/16/2020
 */

import React from 'react';
import ExerciseLog from '../../../src/components/shared/ExerciseLog';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

it('renders correctly', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <ExerciseLog
        user={{username: 'andy', first: 'Andrew', last: 'Jarombek'}}
        addComment={() => {}}
        bucket="all"
        filterBy="all"
        index={0}
        log={{
          log_id: 1,
          username: 'andy',
          first: 'Andrew',
          last: 'Jarombek',
          name: 'Test Log',
          date: '2020-08-26',
          type: 'run',
          distance: 6.15,
          metric: 'miles',
          miles: 6.15,
          time: '00:41:51',
          pace: '00:06:48',
          feel: 7,
          location: 'New York, NY',
          description: 'Reservoir Loop',
          timeCreated: '2020-08-26 12:34:00',
          comments: []
        }}
        newComments={{}}
        page={1}
        postComment={() => {}}
      />
    </MemoryRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
