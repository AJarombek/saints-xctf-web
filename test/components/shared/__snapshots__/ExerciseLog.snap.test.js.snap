// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="exerciseLog-0-1-1 exerciseLog-d0-0-1-22 exerciseLog"
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="headerSection-0-1-2"
  >
    <div
      className="titles-0-1-3"
    >
      <a
        className="titleLink-0-1-4"
        href="/profile/andy"
        onClick={[Function]}
      >
        Andrew
         
        Jarombek
      </a>
      <h6
        className="title-0-1-5"
      >
        Test Log
      </h6>
    </div>
    <div
      className="metadata-0-1-6"
    >
      <p
        className="date-0-1-7"
      >
        Aug. 26th, 2020
      </p>
      <p
        className="type-0-1-8"
      >
        RUN
      </p>
    </div>
  </div>
  <div
    className="bodySection-0-1-9"
  >
    <div
      className="dataFields-0-1-10"
    >
      <p>
        Location: 
        New York, NY
      </p>
      <p>
        6.15
         
        miles
      </p>
      <p>
        41:51
         (
        6:48
        /mi)
      </p>
    </div>
    <div
      className="description-0-1-11"
    >
      Reservoir Loop
    </div>
  </div>
  <div
    className="commentSection-0-1-12"
  >
    <div
      className="comments-0-1-23 comments"
    >
      <div
        className="newCommentForm-0-1-24"
      >
        <textarea
          className="newComment-0-1-25 newComment-d0-0-1-37 blurNewComment-0-1-27"
          disabled={false}
          maxLength={1000}
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Comment"
        />
      </div>
      <div
        className="commentList commentList-0-1-31"
      />
    </div>
  </div>
</div>
`;
